src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/KeyworkerServiceController.kt

```diff
@@ -437,6 +438,31 @@ class KeyworkerServiceController(
   ): OffenderKeyWorkerHistory =
     keyworkerService.getFullAllocationHistory(offenderNo).orElseThrow { EntityNotFoundException() }
 
+  @ApiOperation(
+    value = "Gets a summary of the offender's allocation histories",
+    nickname = "getKeyWorkerHistorySummaryForPrisoners"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        code = 200, message = "OK", response = OffenderKeyWorkerHistory::class
+      ), ApiResponse(
+        code = 400, message = "Invalid request.", response = ErrorResponse::class
+      ), ApiResponse(
+        code = 500,
+        message = "Unrecoverable error occurred whilst processing request.",
+        response = ErrorResponse::class
+      )
+    ]
+  )
+  @PostMapping(path = ["/allocation-history/summary"])
+  fun getKeyWorkerHistorySummaryForPrisoners(
+    @RequestBody offenderNos: List<String>?
+  ): List<OffenderKeyWorkerHistorySummary> {
+    Validate.notEmpty<List<String>?>(offenderNos, "Please provide a list of Offender Nos.")
```

Comment:
Could you handle this using an annotation?

Reply:
I don't know - I copied it from `getOffenderKeyworkerDetailsListPost` as it seemed like a good idea